src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/ConjugatedPiSystemFragmenter.java

```diff
@@ -0,0 +1,304 @@
+        return this.fragmentSaturationSetting;
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="Public Properties Set">
+
+    @Override
+    public void setFragmentSaturationSetting(FragmentSaturationOption anOption) throws NullPointerException {
+        Objects.requireNonNull(anOption, "Given saturation option is null.");
+        this.fragmentSaturationSetting.set(anOption);
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="Public Methods">
+    @Override
+    public IMoleculeFragmenter copy() {
+        ConjugatedPiSystemFragmenter tmpCopy = new ConjugatedPiSystemFragmenter();
+        tmpCopy.setFragmentSaturationSetting((IMoleculeFragmenter.FragmentSaturationOption) this.fragmentSaturationSetting.get());
+        return tmpCopy;
+    }
+
+    @Override
+    public void restoreDefaultSettings() {
+        this.fragmentSaturationSetting.set(IMoleculeFragmenter.FRAGMENT_SATURATION_OPTION_DEFAULT);
+    }
+
+    @Override
+    public List<IAtomContainer> fragmentMolecule(IAtomContainer aMolecule) throws NullPointerException, IllegalArgumentException, CloneNotSupportedException {
+
+        //<editor-fold desc="Molecule Cloning, Property and Arrays Set">
+        IAtomContainer tmpClone = aMolecule.clone();
+        int tmpPreFragmentationCount = 0;
+        for (IAtom tmpAtom: tmpClone.atoms()) {
+            if (tmpAtom.getAtomicNumber() != 0) {
+                tmpPreFragmentationCount++;
+            }
+        }
+        //</editor-fold>
+
+        //<editor-fold desc="Detection and Extraction">
+        IAtomContainerSet tmpConjugatedAtomContainerSet;
+        try {
+            tmpConjugatedAtomContainerSet = ConjugatedPiSystemsDetector.detect(tmpClone);
+        } catch (Exception anException) {
+            ConjugatedPiSystemFragmenter.this.LOGGER.log(Level.WARNING,
+                    anException + " MoleculeID: " + tmpClone.getID(), anException);
+            throw new IllegalArgumentException("Unexpected error occurred during fragmentation of molecule: "
+                    + tmpClone.getID() + " at conjugated pi systems detector: " + anException.toString());
+        }
+        //</editor-fold>
+
+        //<editor-fold desc="Hydrogen Saturation">
+        List<IAtomContainer> tmpProcessedFragments = new ArrayList<>(tmpConjugatedAtomContainerSet.getAtomContainerCount());
+        try {
+            if (!tmpConjugatedAtomContainerSet.isEmpty() && tmpConjugatedAtomContainerSet != null) {
+                CDKHydrogenAdder tmpAdder = CDKHydrogenAdder.getInstance(tmpConjugatedAtomContainerSet.getAtomContainer(0).getBuilder());
+                for (IAtomContainer tmpAtomContainer: tmpConjugatedAtomContainerSet.atomContainers()) {
+                    AtomContainerManipulator.percieveAtomTypesAndConfigureAtoms(tmpAtomContainer);
+                    if (this.fragmentSaturationSetting.get().equals(FragmentSaturationOption.HYDROGEN_SATURATION)) {
```

Comment:
if saturation is not needed, the whole block is not needed, so this check should come higher up.